pkg/infrastructure/infrastructure.go

```diff
@@ -0,0 +1,92 @@
+package infrastructure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	v1 "github.com/openshift/api/config/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	infrastructureName = "cluster"
+)
+
+// Builder provides a struct for infrastructure object from the cluster and a infrastructure definition.
+type Builder struct {
+	// infrastructure definition, used to create the infrastructure object.
+	Definition *v1.Infrastructure
+	// Created infrastructure object.
+	Object *v1.Infrastructure
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+}
```

Comment:
Don't we want to have errorMsg ?

Reply:
I don't see where we could use it in this case

Reply:
Currently it seems not needed. Still was related with the `Update()` discussion.